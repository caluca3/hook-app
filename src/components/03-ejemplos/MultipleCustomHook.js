import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';

import './styles.css';

export const MultipleCustomHook = () => {

  const {counter,increment}=useCounter(1);

  const {loading,data}= useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
  //para una condicion logica sencilla  y convetti booleanos
  //se puede realiuzar ejemplo!!data&&data[0]
  // para efectos necesarios
console.log(data);
  const {author,quote} = !!data && data[0];//si la data existe extrae la posicion 0
  
  
 
  return (
    <>
    <div>
        <h1>BreakingbBad quotes</h1>
        <hr/>
        
         {
                loading?(
        <div className='alert alert-info text-center'>
            loading...
        </div>

                ):
        <blockquote className='text-end'> 
            <p className='mb-0'>{quote}</p>
            <br/>
            <footer className='blockquote-footer'>{author}</footer>
        </blockquote>
         }

        {
        !loading 
         &&
 
            ( 
                <button id='btn' className="btn btn-primary"onClick={ increment }>
                    Siguiente quote
                </button>
            )
        }

    </div>
    </>
  );
}
