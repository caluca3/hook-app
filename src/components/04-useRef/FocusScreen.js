import React, { useRef } from 'react';
import '../03-ejemplos/styles.css';


export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick=()=>{
    inputRef.current.select();
    console.log(inputRef);
}


  return (<>
  
    <div>FocusScreen</div>
    <hr/>

    <input ref={inputRef}
        className='form-control'
        placeholder='Su nombre'
    />

    <button 
        className='btn btn-outline-warning mt-3'
        onClick={handleClick}
    > 
    Focus
    </button>
    
  </>
  )
}
