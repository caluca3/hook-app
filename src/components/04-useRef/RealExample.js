import React, { useState } from 'react'
import { MultipleCustomHook } from '../03-ejemplos/MultipleCustomHook';
import '../03-ejemplos/styles.css';

export const RealExample = () => {


const [show, setShow] = useState(false);
const ocultar = ()=>{
  setShow(!show);
}
  return (<>
  
    <div>RealExample</div>
    <hr/>
{

   show && < MultipleCustomHook/>
}
<button className='btn btn-warning mt-5'
        onClick={ocultar}
 >Show/Hide</button>
     
  </>
  )
}
