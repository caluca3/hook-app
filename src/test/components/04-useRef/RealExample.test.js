import { act, renderHook } from "@testing-library/react-hooks";
import { MultipleCustomHook } from "../../../components/03-ejemplos/MultipleCustomHook";
import { RealExample } from "../../../components/04-useRef/RealExample";


describe('Pruebas en el componente RealExample',()=>{

    const { result } = renderHook(() =>  RealExample());
   
    test('Debe mostrarse', () => { 

  expect(result).toMatchSnapshot();
    });

    test('Debe ocultarse con la funcion', () => {
        const {customHook} = result.all.find('MultipleCustomHook')
        const {botton} = result.current

        act(()=>{
            botton()
        })
       // expect(customHook.)
        expect(customHook).toBe(true);
    });
});