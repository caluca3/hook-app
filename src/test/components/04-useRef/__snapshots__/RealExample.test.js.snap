// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RealExample Debe mostrarse 1`] = `
Object {
  "all": Array [
    <React.Fragment>
      <div>
        RealExample
      </div>
      <hr />
      <button
        className="btn btn-warning mt-5"
        onClick={[Function]}
      >
        Show/Hide
      </button>
    </React.Fragment>,
  ],
  "current": <React.Fragment>
    <div>
      RealExample
    </div>
    <hr />
    <button
      className="btn btn-warning mt-5"
      onClick={[Function]}
    >
      Show/Hide
    </button>
  </React.Fragment>,
  "error": undefined,
}
`;
